### PR TITLE
feat: migrate runner to async db

### PR DESCRIPTION
**Pull request by skyzh**
Signed-off-by: Alex Chi <iskyzh@gmail.com>

finish remaining works of https://github.com/singularity-data/sqllogictest-rs/pull/2

### CODE REVIEW

**Comment by skyzh** [comment]
Note that this patch results in API change `&self` -> `&mut self`, so the next version should be tagged as v0.3.

File: src/runner.rs [hunk 1]
```diff
@@ -203,32 +217,57 @@ impl<D: DB> Runner<D> {
         Ok(())
     }
 
+    /// Run a single record.
+    pub async fn run(&mut self, record: Record) -> Result<(), TestError> {
```
Comment (wangrunji0408) [inline src/runner.rs:221]: This function is intended to be not async?
```suggestion
    pub fn run(&mut self, record: Record) -> Result<(), TestError> {
```

Comment (skyzh) [reply src/runner.rs:221]: Yes, my fault :(

**Review by wangrunji0408 — APPROVED** [review]
LGTM!

**Pull request merged**